### PR TITLE
support search by fullname without space#67

### PR DESCRIPTION
**Pull request by prazuma**
http://personfindersandayo.appspot.com/global/home.html

### CODE REVIEW

File: app/full_text_search.py [hunk 1]
```diff
@@ -131,6 +131,22 @@ def create_jp_name_fields(**kwargs):
                         value=romanized_japanese_name)
                 )
                 romanized_names_list.append(romanized_japanese_name)
+                kwargs[field] = romanized_japanese_name
+
+    # field for searching by fullname without space
```
Comment (gimite) [inline app/full_text_search.py]: "fields for searching by full name without white space"

File: app/full_text_search.py [hunk 2]
```diff
@@ -131,6 +131,22 @@ def create_jp_name_fields(**kwargs):
...
+    romanized_given_name = kwargs['given_name']
+    romanized_family_name = kwargs['family_name']
+    if romanized_given_name and romanized_family_name:
```
Comment (gimite) [inline app/full_text_search.py]: As discussed offline, please apply this for unicode-based romanization too, so that it works for hiragana too.

File: app/full_text_search.py [hunk 3]
```diff
@@ -131,6 +131,22 @@ def create_jp_name_fields(**kwargs):
                         value=romanized_japanese_name)
                 )
                 romanized_names_list.append(romanized_japanese_name)
+                kwargs[field] = romanized_japanese_name
```
Comment (gimite) [inline app/full_text_search.py]: In general, you should avoid modifying method parameters. Can you define another variable to store this?

Comment (prazuma) [inline app/full_text_search.py]: done

File: app/full_text_search.py (hunk 1 above)
Comment (prazuma) [inline app/full_text_search.py]: done

File: app/full_text_search.py (hunk 2 above)
Comment (prazuma) [inline app/full_text_search.py]: done

File: app/full_text_search.py [hunk 4]
```diff
@@ -131,6 +145,27 @@ def create_jp_name_fields(**kwargs):
                         value=romanized_japanese_name)
                 )
                 romanized_names_list.append(romanized_japanese_name)
+                romanized_name_values[field] = romanized_japanese_name
+        else:
+            romanized_name_values[field] = kwargs[field]
+
+    # fields for searching by full name without white space
```
Comment (gimite) [inline app/full_text_search.py]: As discussed, please avoid repeating this code twice.

Comment (prazuma) [inline app/full_text_search.py]: done

File: app/full_text_search.py [hunk 5]
```diff
@@ -113,12 +113,55 @@ def search(repo, query_txt, max_results):
...
+    """
+    full_name_without_space = ''
+    if given_name and family_name:
+        full_name_without_space = given_name + family_name
+    return full_name_without_space
+
+
+def create_full_name_without_space_fields(romanize_method, given_name, family_name):
+    """
+    Creates full name without fields.
```
Comment (gimite) [inline app/full_text_search.py]: "Creates fields with the full name without white spaces."

Comment (prazuma) [inline app/full_text_search.py]: done

File: app/full_text_search.py [hunk 6]
```diff
@@ -158,6 +213,7 @@ def create_jp_location_fields(**kwargs):
                 )
     return fields
 
+
 def create_document(record_id, repo, **kwargs):
```
Comment (gimite) [inline app/full_text_search.py]: As discussed, please refactor the code to create a function similar to create_jp_name_fields which supports both romanization.

Comment (prazuma) [inline app/full_text_search.py]: done

File: app/script_variant.py [hunk 7]
```diff
@@ -54,9 +55,10 @@ def romanize_japanese_location(word):
     if not word:
         return word
 
-    if word in JAPANESE_LOCATION_DICTIONARY:
-        yomigana = JAPANESE_LOCATION_DICTIONARY[word]
-        return jautils.hiragana_to_romaji(yomigana)
+    if (re.match(ur'([\u3400-\u9fff])', word)):
```
Comment (gimite) [inline app/script_variant.py]: This line appears twice. Can you extract this line into a function?

Also, can you add a comment what is [\u3400-\u9fff]?

File: app/full_text_search.py [hunk 8]
```diff
@@ -170,21 +215,35 @@ def create_document(record_id, repo, **kwargs):
     fields.append(appengine_search.TextField(name='repo', value=repo))
     fields.append(appengine_search.TextField(name='record_id', value=record_id))
 
-    # Add name and location romanized by unidecode
-    for field in kwargs:
-        romanized_value = script_variant.romanize_word(kwargs[field])
-        fields.append(
-            appengine_search.TextField(name=field, value=romanized_value))
+    # Add name romanized by unidecode
+    fields.extend(create_romanized_name_fields(
```
Comment (gimite) [inline app/full_text_search.py]: You have two similar calls of create_romanized_name_fields and create_romanized_location_fields. Can you simplify the code here using for-loop?

File: app/full_text_search.py [hunk 9]
```diff
@@ -192,6 +251,7 @@ def create_document(record_id, repo, **kwargs):
         home_neighborhood=kwargs['home_neighborhood'],
         home_country=kwargs['home_country']))
 
+    logging.info(fields)
```
Comment (gimite) [inline app/full_text_search.py]: Remove this line.

File: tests/test_full_text_search.py [hunk 10]
```diff
@@ -192,6 +202,12 @@ def test_search_by_name_only(self):
         assert set([r.record_id for r in results]) == \
             set(['haiti/0225'])
 
+        # Search by full name without space
```
Comment (gimite) [inline tests/test_full_text_search.py]: Also add a unit test to test full name without space + romanization applied (both unicode one and Japanese dictionary one). My guess is that the test fails :)

File: app/full_text_search.py [hunk 11]
```diff
@@ -113,51 +116,93 @@ def search(repo, query_txt, max_results):
     return results
 
 
-def create_jp_name_fields(**kwargs):
+def create_full_name_without_space(given_name, family_name):
```
Comment (gimite) [inline app/full_text_search.py]: You may alternatively return a tuple: (given_name + family_name, family_name + given_name).

Otherwise, rename given_name, family_name here to name_component1, name_component2 or something, because currently family_name may be passed as given_name, which is confusing.

File: app/full_text_search.py [hunk 12]
```diff
@@ -113,51 +116,93 @@ def search(repo, query_txt, max_results):
...
+            romanized_names_list.append(romanized_name)
+
+    full_name_field, romanized_full_names = create_full_name_without_space_fields(
+        romanize_method, kwargs['given_name'], kwargs['family_name'])
+    fields.extend(full_name_field)
+    romanized_names_list.extend(romanized_full_names)
+
+    names = ':'.join([name for name in romanized_names_list if name])
+    fields.append(
+        appengine_search.TextField(name='name_'+romanize_method_name,
```
Comment (gimite) [inline app/full_text_search.py]: Line 173 generates {field}_romanized_by_{method_name}, but here generates name_{method_name}. Can you be consistent with the naming?

Also, the field name here should be "names", not "name".

File: app/full_text_search.py [hunk 13]
```diff
@@ -113,51 +116,93 @@ def search(repo, query_txt, max_results):
...
+    romanized_family_name = romanize_method(family_name)
+    full_name_given_family = create_full_name_without_space(
+        romanized_given_name, romanized_family_name
+    )
+    full_name_family_given = create_full_name_without_space(
+        romanized_family_name, romanized_given_name
+    )
+    if full_name_given_family:
+        fields.append(appengine_search.TextField(
+            name='no_spacefull_name_1_'+romanize_method.__name__,
```
Comment (gimite) [inline app/full_text_search.py]: Should be "no_space_full_name_1_" (with one more underscore).

File: app/full_text_search.py [hunk 14]
```diff
@@ -113,51 +116,93 @@ def search(repo, query_txt, max_results):
...
+            value=full_name_given_family))
+        romanized_name_list.append(full_name_given_family)
+    if full_name_family_given:
+        fields.append(appengine_search.TextField(
+            name='no_spacefull_name_2_'+romanize_method.__name__,
```
Comment (gimite) [inline app/full_text_search.py]: Ditto.

File: app/full_text_search.py [hunk 15]
```diff
@@ -170,21 +215,35 @@ def create_document(record_id, repo, **kwargs):
...
+    fields.extend(create_romanized_name_fields(
+        script_variant.romanize_word,
+        given_name=kwargs['given_name'],
+        family_name=kwargs['family_name'],
+        full_name=kwargs['full_name'],
+        alternate_names=kwargs['alternate_names']))
+
+    # Add location romanized by unidecode
+    fields.extend(create_romanized_location_fields(
+        script_variant.romanize_word,
```
Comment (gimite) [inline app/full_text_search.py]: "romanize_word" sounds vague in contrast to romanize_japanese_name_by_name_dict etc. Can you rename it to something like "romanize_word_by_unidecode"?

File: app/full_text_search.py [hunk 16]
```diff
@@ -113,51 +116,93 @@ def search(repo, query_txt, max_results):
...
-                                             value=romanized_jp_names))
+        romanized_name = romanize_method(kwargs[field])
+        if romanized_name:
+            fields.append(
+                appengine_search.TextField(
+                    name=field+'_romanized_by_'+romanize_method_name,
+                    value=romanized_name))
+            romanized_names_list.append(romanized_name)
+
+    full_name_field, romanized_full_names = create_full_name_without_space_fields(
```
Comment (gimite) [inline app/full_text_search.py]: Please use plural form as a variable name if the variable stores an array.

File: app/full_text_search.py [hunk 17]
```diff
@@ -113,51 +116,93 @@ def search(repo, query_txt, max_results):
...
+                                   value=names))
     return fields
 
 
-def create_jp_location_fields(**kwargs):
+def create_romanized_location_fields(romanize_method, **kwargs):
     """
-    Creates fields(romanized jp location data) for full text search.
-    It should be called in create_document method.
+    Creates fields(romanized by romanize_method) for full text search.
```
Comment (gimite) [inline app/full_text_search.py]: This comment is exactly the same as line 163. Can you update both to clarify the difference?

File: app/full_text_search.py [hunk 18]
```diff
@@ -113,51 +116,93 @@ def search(repo, query_txt, max_results):
...
+        romanized_japanese_location = romanize_method(kwargs[field])
+        if romanized_japanese_location:
+            fields.append(
+                appengine_search.TextField(
+                    name=field+'_romanized_by_'+romanize_method_name,
+                    value=romanized_japanese_location)
+            )
     return fields
 
+
```
Comment (gimite) [inline app/full_text_search.py]: Can you remove this extra empty line?

File: app/full_text_search.py [hunk 19]
```diff
@@ -204,15 +264,9 @@ def add_record_to_index(person):
     """
     person_location_index = appengine_search.Index(
         name=PERSON_LOCATION_FULL_TEXT_INDEX_NAME)
-    name_params = [person.given_name,
-                   person.family_name,
-                   person.full_name,
-                   person.alternate_names]
-    names =  ':'.join([name for name in name_params if name])
     person_location_index.put(create_document(
```
Comment (gimite) [inline app/full_text_search.py]: How about passing `person` object to create_document instead of the set of fields? Anyway you select the fields in create_document, right?

File: tests/test_full_text_search.py [hunk 20]
```diff
@@ -192,6 +202,12 @@ def test_search_by_name_only(self):
         assert set([r.record_id for r in results]) == \
```
Comment (gimite) [inline tests/test_full_text_search.py]: Do you have equivalent unit test using romanization with unidecode? Please add one if missing.

File: app/full_text_search.py (hunk 11 above)
Comment (prazuma) [inline app/full_text_search.py]: done

File: app/full_text_search.py (hunk 13 above)
Comment (prazuma) [inline app/full_text_search.py]: done

File: app/full_text_search.py (hunk 14 above)
Comment (prazuma) [inline app/full_text_search.py]: done

File: app/full_text_search.py (hunk 16 above)
Comment (prazuma) [inline app/full_text_search.py]: done

File: app/full_text_search.py (hunk 12 above)
Comment (prazuma) [inline app/full_text_search.py]: doen

File: app/full_text_search.py (hunk 17 above)
Comment (prazuma) [inline app/full_text_search.py]: done

File: app/full_text_search.py (hunk 18 above)
Comment (prazuma) [inline app/full_text_search.py]: done

File: app/full_text_search.py (hunk 9 above)
Comment (prazuma) [inline app/full_text_search.py]: done

File: app/full_text_search.py (hunk 8 above)
Comment (prazuma) [inline app/full_text_search.py]: done

File: app/full_text_search.py (hunk 15 above)
Comment (prazuma) [inline app/full_text_search.py]: romanize_word -> romanize_word_by_unidecode

File: app/full_text_search.py (hunk 19 above)
Comment (prazuma) [inline app/full_text_search.py]: done

File: app/script_variant.py (hunk 7 above)
Comment (prazuma) [inline app/script_variant.py]: done

File: tests/test_full_text_search.py (hunk 20 above)
Comment (prazuma) [inline tests/test_full_text_search.py]: done

File: tests/test_full_text_search.py (hunk 10 above)
Comment (prazuma) [inline tests/test_full_text_search.py]: done

File: app/script_variant.py [hunk 21]
```diff
@@ -29,6 +29,14 @@ def read_dictionary(file_name):
 JAPANESE_LOCATION_DICTIONARY = read_dictionary('jp_location_dict.txt')
 
 
+def is_kanji(word):
+    """
+    Returns word is kanji or not.
```
Comment (gimite) [inline app/script_variant.py]: Returns *whether* word is kanji or not.

File: app/script_variant.py [hunk 22]
```diff
@@ -29,6 +29,14 @@ def read_dictionary(file_name):
...
+    [\u3400-\u9fff]: kanji
+    """
+    return re.match(ur'([\u3400-\u9fff])', word)
```
Comment (gimite) [inline app/script_variant.py]: This range is actually not very accurate. Some of kanji characters are out of this range. Can you add TODO to update this to contains all kanji characters?

File: app/script_variant.py (hunk 21 above)
Comment (prazuma) [inline app/script_variant.py]: done

File: tests/test_full_text_search.py [hunk 23]
```diff
@@ -192,6 +221,27 @@ def test_search_by_name_only(self):
...
+        results = full_text_search.search('haiti', u'ちはや あらお', 5)
+        assert set([r.record_id for r in results]) == \
+            set(['haiti/0225'])
+
+        # Search by full name without space
+        results = full_text_search.search('haiti', 'HibikiGanaha', 5)
+        assert set([r.record_id for r in results]) == \
+            set(['haiti/1010'])
+
+        # Search by full name without space
```
Comment (gimite) [inline tests/test_full_text_search.py]: Can you update comment here and line 239, to clarify how these are different from the previous test?

File: app/script_variant.py (hunk 22 above)
Comment (prazuma) [inline app/script_variant.py]: added todo comment

File: app/full_text_search.py [hunk 24]
```diff
@@ -113,52 +116,93 @@ def search(repo, query_txt, max_results):
     return results
 
 
-def create_jp_name_fields(**kwargs):
+def create_full_name_without_space(name_component1, name_component2):
```
Comment (gimite) [inline app/full_text_search.py]: If you return a tuple, then naming given_name, family_name is better because you always pass these two in this order :)

File: app/full_text_search.py [hunk 25]
```diff
@@ -113,52 +116,93 @@ def search(repo, query_txt, max_results):
...
     """
-    Creates fields(romanized_jp_names) for full text search.
+    Creates full name without white space.
+    Returns:
+        if name_component1 and name_component2: 'name_component1 + name_component2'
```
Comment (gimite) [inline app/full_text_search.py]: Please update this comment because it now returns a tuple.

File: app/full_text_search.py [hunk 26]
```diff
@@ -113,52 +116,93 @@ def search(repo, query_txt, max_results):
...
+        else: None
+    """
+    full_name_without_space = ''
```
Comment (gimite) [inline app/full_text_search.py]: The comment says it returns None, but actually it returns ''. Please fix.

File: app/full_text_search.py [hunk 27]
```diff
@@ -113,52 +116,93 @@ def search(repo, query_txt, max_results):
...
+    if name_component1 and name_component2:
```
Comment (gimite) [inline app/full_text_search.py]: It would be clearer to write:

```
if ...:
  return ...
else:
  return ...
```

File: app/full_text_search.py [hunk 28]
```diff
@@ -113,52 +116,93 @@ def search(repo, query_txt, max_results):
...
+        fullname fields, romanized_name_list: (for check)
+    """
+    fields = []
+    romanized_name_list = []
+    romanized_given_name = romanize_method(given_name)
+    romanized_family_name = romanize_method(family_name)
+    full_names = create_full_name_without_space(
+        romanized_given_name, romanized_family_name)
+    if full_names:
+        full_name_given_family = full_names[0]
```
Comment (gimite) [inline app/full_text_search.py]: Optional but this may be simpler:

    full_name_given_family, full_name_family_given = full_names

File: tests/test_full_text_search.py (hunk 23 above)
Comment (prazuma) [inline tests/test_full_text_search.py]: done

File: app/full_text_search.py [hunk 29]
```diff
@@ -113,52 +116,93 @@ def search(repo, query_txt, max_results):
...
+        full_name_family_given = full_names[1]
+        fields.append(appengine_search.TextField(
+            name='no_space_full_name_1_'+romanize_method.__name__,
```
Comment (gimite) [inline app/full_text_search.py]: Again, can you be consistent with the naming? This field should also be named ...romanized_by...

File: app/full_text_search.py (hunk 24 above)
Comment (prazuma) [inline app/full_text_search.py]: done

File: app/full_text_search.py (hunk 25 above)
Comment (prazuma) [inline app/full_text_search.py]: done

File: app/full_text_search.py (hunk 26 above)
Comment (prazuma) [inline app/full_text_search.py]: done

File: app/script_variant.py [hunk 30]
```diff
@@ -54,14 +63,15 @@ def romanize_japanese_location(word):
...
+    if is_kanji(word):
+        if word in JAPANESE_LOCATION_DICTIONARY:
+            yomigana = JAPANESE_LOCATION_DICTIONARY[word]
+            return jautils.hiragana_to_romaji(yomigana)
 
     return word
 
 
-def romanize_word(word):
+def romanize_word_by_unidecode(word):
```
Comment (gimite) [inline app/script_variant.py]: As discussed, this actually also applies Japanese romanization too. Please fix.

File: app/full_text_search.py (hunk 27 above)
Comment (prazuma) [inline app/full_text_search.py]: done

File: app/full_text_search.py (hunk 28 above)
Comment (prazuma) [inline app/full_text_search.py]: done

File: app/full_text_search.py [hunk 31]
```diff
@@ -113,52 +116,93 @@ def search(repo, query_txt, max_results):
...
+                                   value=names))
     return fields
 
 
-def create_jp_location_fields(**kwargs):
+def create_romanized_location_fields(romanize_method, **kwargs):
     """
-    Creates fields(romanized jp location data) for full text search.
-    It should be called in create_document method.
+    Creates romanized location fields(romanized by romanize_method)
```
Comment (gimite) [inline app/full_text_search.py]: Nit: Put whitespace before "(".

File: app/full_text_search.py (hunk 29 above)
Comment (prazuma) [inline app/full_text_search.py]: done

File: app/script_variant.py (hunk 30 above)
Comment (prazuma) [inline app/script_variant.py]: done

File: app/full_text_search.py (hunk 31 above)
Comment (prazuma) [inline app/full_text_search.py]: done

File: app/script_variant.py [hunk 32]
```diff
@@ -29,6 +29,15 @@ def read_dictionary(file_name):
 JAPANESE_LOCATION_DICTIONARY = read_dictionary('jp_location_dict.txt')
 
 
+def is_kanji(word):
+    """
+    Returns whether word is kanji or not.
+    [\u3400-\u9fff]: kanji
```
Comment (gimite) [inline app/script_variant.py]: Nit: Please move this comment (about [\u3400-\u9fff]) out of docstring ("""...""") and make it a normal comment "#...". Docstring should comment the spec (what is this function and how to use it), not its internal implementation.

File: app/script_variant.py [hunk 33]
```diff
@@ -54,14 +63,15 @@ def romanize_japanese_location(word):
...
+            return jautils.hiragana_to_romaji(yomigana)
 
     return word
 
 
-def romanize_word(word):
+def romanize_word_by_unidecode(word):
     """
     This method romanizes all languages by unidecode.
     If word is hiragana or katakana, it is romanized by jautils.
```
Comment (gimite) [inline app/script_variant.py]: Can you add a comment that this function romanizes kanji in Chinese way?

File: app/script_variant.py [hunk 34]
```diff
@@ -54,14 +63,15 @@ def romanize_japanese_location(word):
     if not word:
         return word
 
-    if word in JAPANESE_LOCATION_DICTIONARY:
-        yomigana = JAPANESE_LOCATION_DICTIONARY[word]
-        return jautils.hiragana_to_romaji(yomigana)
+    if is_kanji(word):
```
Comment (gimite) [inline app/script_variant.py]: As discussed, please romanize hiragana too here, to be consistent with romanize_japanese_name_by_name_dict.

File: app/script_variant.py (hunk 32 above)
Comment (prazuma) [inline app/script_variant.py]: done

File: app/script_variant.py [hunk 35]
```diff
@@ -29,6 +29,15 @@ def read_dictionary(file_name):
...
+    Returns whether word is kanji or not.
+    [\u3400-\u9fff]: kanji
+    TODO: update this range (some of kanji characters out of this range.)
+    """
+    return re.match(ur'([\u3400-\u9fff])', word)
+
+
 def romanize_japanese_name_by_name_dict(word):
     """
     This method romanizes japanese name by using name dictionary.
```
Comment (gimite) [inline app/script_variant.py]: > If word isn't found in dictionary, this method doesn't apply romanize.

This is not exactly true. You romanize hiragana words not in dictionary, right? Please fix.

File: app/script_variant.py (hunk 33 above)
Comment (prazuma) [inline app/script_variant.py]: done

File: app/script_variant.py (hunk 34 above)
Comment (prazuma) [inline app/script_variant.py]: done

File: app/script_variant.py (hunk 35 above)
Comment (gimite) [inline app/script_variant.py]: As chatted, I read the code in a wrong way. Please ignore.

File: app/script_variant.py (hunk 34 above)
Comment (gimite) [inline app/script_variant.py]: As chatted, I read the code in a wrong way. Please ignore.

File: app/full_text_search.py [hunk 36]
```diff
@@ -113,52 +116,93 @@ def search(repo, query_txt, max_results):
...
+    Returns:
+        if given_name and family_name: ('given_name + family_name',
+                                        'family_name + given_name')
+        else: ''
+    """
+    if given_name and family_name:
+        return (given_name + family_name,
+                family_name + given_name)
+    else:
+        return ''
```
Comment (gimite) [inline app/full_text_search.py]: As chatted, please return None instead.

Comment (prazuma) [inline app/full_text_search.py]: done

File: app/script_variant.py (hunk 34 above)
Comment (gimite) [inline app/script_variant.py]: By the way, why do you have is_kanji here but not in romanize_japanese_name_by_name_dict?

File: app/full_text_search.py [hunk 37]
```diff
@@ -166,31 +210,34 @@ def create_document(record_id, repo, **kwargs):
...
-
-    # Add location romanized by japanese location dictionary
-    fields.extend(create_jp_location_fields(
-        home_street=kwargs['home_street'],
-        home_city=kwargs['home_city'],
-        home_state=kwargs['home_state'],
-        home_postal_code=kwargs['home_postal_code'],
-        home_neighborhood=kwargs['home_neighborhood'],
-        home_country=kwargs['home_country']))
+    romanize_name_methods = [script_variant.romanize_word_by_unidecode,
```
Comment (gimite) [inline app/full_text_search.py]: Please comment why you need to apply two methods.

Comment (prazuma) [inline app/full_text_search.py]: done

File: app/script_variant.py (hunk 34 above)
Comment (prazuma) [inline app/script_variant.py]: fixed

File: app/script_variant.py [hunk 38]
```diff
@@ -29,6 +29,15 @@ def read_dictionary(file_name):
 JAPANESE_LOCATION_DICTIONARY = read_dictionary('jp_location_dict.txt')
 
 
+def is_kanji(word):
```
Comment (gimite) [inline app/script_variant.py]: Actually this should be named has_kanji, and update the comment that it return whether the word *contains* kanji or not.

A word can contain both kanji and non-kanji e.g., "のび太" and it returns true for such word in the current implementation.

File: app/full_text_search.py [hunk 39]
```diff
@@ -166,31 +210,37 @@ def create_document(record_id, repo, **kwargs):
...
-
-    # Add location romanized by japanese location dictionary
-    fields.extend(create_jp_location_fields(
-        home_street=kwargs['home_street'],
-        home_city=kwargs['home_city'],
-        home_state=kwargs['home_state'],
-        home_postal_code=kwargs['home_postal_code'],
-        home_neighborhood=kwargs['home_neighborhood'],
-        home_country=kwargs['home_country']))
+    # romanize name method by unidecode or japanese name dictionary.
```
Comment (gimite) [inline app/full_text_search.py]: Please explain why, instead of what you are doing, which is obvious from code in this case.

Comment (prazuma) [inline app/full_text_search.py]: done

File: app/script_variant.py (hunk 38 above)
Comment (prazuma) [inline app/script_variant.py]: done

**Comment by gimite** [comment]
LGTM. Please rerun tools/all_tests and merge the pull request if it all succeeds.

**Comment by prazuma** [comment]
all_tests passed(∩´∀｀)∩ﾜｰｲ
